src/test/java/org/folio/rest/impl/OaiPmhImplTest.java

```diff
@@ -2457,27 +2483,33 @@ void verifyResumptionTokenFlow_whenVerbListRecordsAndMetadataPrefixMarc21WithHol
   }
 
   private ResumptionTokenType makeResumptionTokenRequestsAndVerifyCount(List<HeaderType> totalRecords,
-                                                                        ResumptionTokenType resumptionToken, int desiredCount, int expectedCursor) {
-    RequestSpecification request;
+      ResumptionTokenType resumptionToken, VerbType verb, int desiredCount, int expectedCursor) {
+    RequestSpecification request = createBaseRequest().with()
+      .param(VERB_PARAM, verb.value())
+      .param(RESUMPTION_TOKEN_PARAM, resumptionToken.getValue());
     OAIPMH oaipmh;
     List<HeaderType> records;
-    request = createBaseRequest()
-      .with()
-      .param(VERB_PARAM, LIST_RECORDS.value())
-      .param(RESUMPTION_TOKEN_PARAM, resumptionToken.getValue());
-    oaipmh = verify200WithXml(request, LIST_RECORDS);
-    verifyListResponse(oaipmh, LIST_RECORDS, desiredCount);
-    resumptionToken = getResumptionToken(oaipmh, LIST_RECORDS);
+    oaipmh = verify200WithXml(request, verb);
+    verifyListResponse(oaipmh, verb, desiredCount);
+    resumptionToken = getResumptionToken(oaipmh, verb);
     assertThat(resumptionToken, is(notNullValue()));
-    assertEquals(expectedCursor, resumptionToken.getCursor().intValue());
-
-    records = oaipmh.getListRecords().getRecords().stream()
-      .map(RecordType::getHeader)
-      .collect(Collectors.toList());
+    assertEquals(expectedCursor, resumptionToken.getCursor()
+      .intValue());
+    records = getHeadersListDependOnVerbType(verb, oaipmh);
     totalRecords.addAll(records);
     return resumptionToken;
   }
 
+  private List<HeaderType> getHeadersListDependOnVerbType(VerbType verb, OAIPMH oaipmh) {
+    return verb.equals(LIST_RECORDS) ? oaipmh.getListRecords()
+      .getRecords()
+      .stream()
+      .map(RecordType::getHeader)
+      .collect(Collectors.toList()) :
+      oaipmh.getListIdentifiers()
+      .getHeaders();
```

Comment:
```suggestion
    return verb.equals(LIST_RECORDS) 
      ? oaipmh.getListRecords()
      .getRecords()
      .stream()
      .map(RecordType::getHeader)
      .collect(Collectors.toList()) 
      : oaipmh.getListIdentifiers()
      .getHeaders();
```

Usually `?` and `:` are on separate lines for long line